rllib/utils/metrics/metrics_logger.py

```diff
@@ -736,44 +757,17 @@ def build_nested_dict(stats_dict, key):
                 if self._key_in_stats(key, stats=s)
             ]
 
-            # Find a base Stats object first - we will use this to merge with incoming Stats objects
```

Comment:
I'm moving a good chunk of this "merge nested dicts with stats" functionality into a utility function.
That a) encapsulates the logic for readability and b) enables us to test merging without the MetricsLogger.